### ISSUE TITLE
ClinicTrendsAI - Expand Logistic Regression hyperparameter tuning

### ISSUE THREAD

**Issue by lfariabr** [lfariabr/masters-swe-ai#32]
Extend the current model pipeline to perform hyperparameter tuning for Logistic Regression beyond just max_features. This aims to optimize model performance and prevent overfitting.

**Tasks:**

- Implement GridSearchCV for parameters such as:
-- C (regularization strength)
-- penalty (L1 vs. L2)
-- solver (e.g. lbfgs, saga)
-- multi_class (ovr vs multinomial)
- Evaluate impact on accuracy, F1-score, and convergence speed.
- Document the best hyperparameter combinations found.
